### DISCUSSION TITLE
isKeyDown was not found in the current build context

### DISCUSSION THREAD

**Discussion by meganloz** [bonsai-rx/bonsai#2023] [Q&A]
I am trying to run this script that has previously worked. I use KeyDown to initiate the fiber recording and the shock timer, but the first instance of isKeyDown returns this error. Has anyone experienced this?

[bonsai_fearcond.txt](https://github.com/user-attachments/files/17065241/bonsai_fearcond.txt)

![image](https://github.com/user-attachments/assets/7cd1c6e1-2129-4950-8a32-37b8533de469)
![image](https://github.com/user-attachments/assets/005cd809-77a8-4100-bf2e-7498b372c101)
![image](https://github.com/user-attachments/assets/7cb5ee74-373a-4afe-a060-aa1147b0f447)
[bonsai_fearcond.txt](https://github.com/user-attachments/files/17065241/bonsai_fearcond.txt)

**Comment by bruno-f-cruz** [comment]
Heya!

There appears to be a missing subject that is being referenced without being created in the first place. If it was running in its previous state I would advise you to compare and add the missing code.

**Reply by meganloz** [reply]
Thanks! This is the same file I ran previously so I suspect it was modified in some way such that it no longer works. 

I tried to make the adjustment you suggested but I'm finding that the key down is not triggering acquisition as I hoped. I want to start the session and then use key down to simultaneously begin a) recording from 2 cameras via VideoCaptureDevice, b) begin a shock timer such that I know the precise timing of deliveries (timings will be set in Bonsai), and c) trigger laser output and recording. As it is set up now only the laser appears to begin following a key press. I know that the cameras are not under a KeyDown node but I can't seem to integrate this/ create a connection. 

[bonsai condisc.txt](https://github.com/user-attachments/files/17165967/bonsai.condisc.txt)
![image](https://github.com/user-attachments/assets/d0317826-0e20-49d8-a6fd-696e595664b6)
![image](https://github.com/user-attachments/assets/c0e2a729-273f-4918-9b59-df3d16b1ca74)
